config/src/main/java/org/springframework/security/config/annotation/web/configurers/RequestCacheConfigurer.java

```diff
@@ -120,28 +120,19 @@ private RequestCache getRequestCache(H http) {
 		if (result != null) {
 			return result;
 		}
-		result = getBeanOrNull(RequestCache.class);
-		if (result != null) {
-			return result;
+
+		var context = getBuilder().getSharedObject(ApplicationContext.class);
+		if (context != null) {
```

Comment:
Spring Security tends to avoid nesting conditionals. Even though this is the only place in the file, will you please continue to use `getBeanOrNull`, simply changing that to use `ObjectProvider`? In this way, the `context` and `result` null checks won't be nested.